### PR TITLE
Secure Admin Accounts: Allow only google oauth for admins

### PR DESCRIPTION
**Pull request by nkiruka**
# Description
Purpose of this PR is to merge changes in "allow-only_google-oauth-for-admin" branch onto staging branch.  Initially, the changes were merged into  ["refactor_enforce_google_sso_method"](https://github.com/code-dot-org/code-dot-org/pull/31847) branch

### CODE REVIEW

**Comment by nkiruka** [comment]
There are tests that should not be failing

File: dashboard/app/models/authentication_option.rb [hunk 1]
```diff
@@ -62,6 +62,10 @@ class AuthenticationOption < ApplicationRecord
     MICROSOFT
   ]
 
+  def google_oauth2?
```
Comment (Hamms) [inline dashboard/app/models/authentication_option.rb]: I'd call this either 
```suggestion
  def google_oauth?
```

or just 
```suggestion
  def google?
```

File: dashboard/test/models/user_test.rb [hunk 2]
```diff
@@ -4140,51 +4140,95 @@ def hide_scripts_in_sections(section1, section2)
...
-    assert_equal ["Admin email must have code.org domain"], migrated_teacher.errors.full_messages
+    assert_equal ["Admin must be a code.org account with only google oauth"], migrated_teacher.errors.full_messages
   end
 
-  test 'cannot grant admin role when unmigrated teacher account' do
-    unmigrated_teacher_without_password = create :teacher, :demigrated
-    unmigrated_teacher_without_password.update_attribute(:encrypted_password, '')
+  test 'can grant admin role when in development environment' do
+    with_rack_env(:development) do
+      email = 'katherinejohnson@code.org'
```
Comment (Hamms) [inline dashboard/test/models/user_test.rb:4215]: :rocket:

File: dashboard/test/models/user_test.rb [hunk 3]
```diff
@@ -4140,51 +4140,95 @@ def hide_scripts_in_sections(section1, section2)
...
+      assert migrated_teacher.reload.admin?
     end
+  end
 
-    refute unmigrated_teacher_without_password.reload.admin?
-    assert unmigrated_teacher_without_password.errors[:admin].length == 3
-    assert_equal ["Admin must be a migrated user", "Admin must have Google OAuth", "Admin email must have code.org domain"], unmigrated_teacher_without_password.errors.full_messages
+  test 'can grant admin role when in adhoc environment' do
+    with_rack_env(:adhoc) do
+      email = 'dorothyvaughan@code.org'
```
Comment (Hamms) [inline dashboard/test/models/user_test.rb:4226]: :rocket:

**Review by Hamms — COMMENTED** [review]
Love the new tests!

Looking at some of the tests that are failing, it looks like they're trying to create and use admin users for testing purposes.

I think you'll need to update https://github.com/code-dot-org/code-dot-org/blob/882f23280dcc9cae26751694603594809965a1d0/dashboard/test/factories/factories.rb#L69-L72 to use `google_sso_provider` instead of `with_google_authentication_option` like you did for your own test.

**Review by Hamms — APPROVED** [review]

**Pull request merged**